src/storage/client/StorageClient.cpp

```diff
@@ -454,5 +498,27 @@ StorageClient::get(GraphSpaceID space,
                               });
 }
 
+StatusOr<std::unordered_map<std::pair<GraphSpaceID, PartitionID>, HostAddr>>
+StorageClient::preHeatLeaders() {
+    return getLeaders().via(ioThreadPool_->getEventBase()).thenValue([](auto&& leaders){
+        if (UNLIKELY(!leaders.succeeded())) {
+            LOG(WARNING) << "At least one RPC call failed";
+        }
+        std::unordered_map<std::pair<GraphSpaceID, PartitionID>, HostAddr> r;
+        for (auto& leader : leaders.responses()) {
+            for (auto& val : leader.second.get_leader_parts()) {
+                for (auto& p : val.second) {
+                    r.emplace(std::make_pair(val.first, p), leader.first);
+                }
+            }
+        }
+        return decltype(preHeatLeaders())(std::move(r));
+    }).thenError([](auto&& e){
+        LOG(ERROR) << "Get leaders failed: " << e.what();
+        return decltype(preHeatLeaders())();
```

Comment:
why not return Status("some error message")?

Reply:
Thanks.I should return error message.